services/web/server/src/simcore_service_webserver/folders/_folders_db.py

```diff
@@ -0,0 +1,223 @@
+""" Database API
+
+    - Adds a layer to the postgres API with a focus on the projects comments
+
+"""
+
+import logging
+from typing import cast
+
+from aiohttp import web
+from models_library.folders import FolderDB, FolderID
+from models_library.products import ProductName
+from models_library.rest_ordering import OrderBy, OrderDirection
+from models_library.users import GroupID, UserID
+from models_library.workspaces import WorkspaceID
+from pydantic import NonNegativeInt, parse_obj_as
+from simcore_postgres_database.models.folders_v2 import folders_v2
+from sqlalchemy import func
+from sqlalchemy.sql import asc, desc, select
+
+from ..db.plugin import get_database_engine
+from .errors import FolderAccessForbiddenError, FolderNotFoundError
+
+_logger = logging.getLogger(__name__)
+
+
+_SELECTION_ARGS = (
+    folders_v2.c.folder_id,
+    folders_v2.c.name,
+    folders_v2.c.parent_folder_id,
+    folders_v2.c.created_by_gid,
+    folders_v2.c.created,
+    folders_v2.c.modified,
+    folders_v2.c.user_id,
+    folders_v2.c.workspace_id,
+)
+
+
+async def create_folder(
+    app: web.Application,
+    product_name: ProductName,
```

Comment:
MINOR: some comments about the repo API function signatures:

- I would restrict `engine` as first argument . This way you enforce that it is all you can use from the app. Note that when you pass an app, the function has access to a lot of stuff and in addition the `app` can be in different states
- enforce kwargs, i.e. `app, *, product_name ,...`
- I like to sort the input parameters to follow a particular context, e.g.  `user` `product` define a context that determines `a user in a product`.  This way you are documenting your intentions directly in the code.

Reply:
improved